### PR TITLE
updated pytorch version to 2.2

### PR DESCRIPTION
**Pull request by johnbensnyder**
*Issue #, if available:*

This fixes https://github.com/aws-samples/awsome-distributed-training/issues/219 and https://github.com/aws-samples/awsome-distributed-training/issues/220

*Description of changes:*

- Updates PyTorch version to 2.2. 
- Fixes `triu_tril_cuda_template` error. 
- Updates checkpointing to be compliant with PyTorch >=2.1.
- Changes `sharding_strategy` to default to `full` which matches behavior of previous versions.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: 3.test_cases/10.FSDP/2.distributed-training-mistral.sbatch [hunk 1]
```diff
@@ -28,6 +28,8 @@ export FI_EFA_FORK_SAFE=1
 export FI_LOG_LEVEL=1
 export FI_PROVIDER=efa
 export NCCL_DEBUG=INFO
+## Switching SYNC_MEMOPS to zero can boost throughput on P5
```
Comment (sean-smith) [inline 3.test_cases/10.FSDP/2.distributed-training-mistral.sbatch:31]: Let's link to a relevant GH issue / docs on why we added this.

File: 3.test_cases/10.FSDP/1.distributed-training.sbatch [hunk 2]
```diff
@@ -59,14 +61,15 @@ declare -a TRAINING_ARGS=(
...
-    --max_steps 5000 \
+    --max_steps=5000 \
     --checkpoint_dir=./checkpoints \
     --dataset='c4' \
     --dataset_config_name='en' \
     --resume_from_checkpoint=./checkpoints \
     --train_batch_size=1 \
     --val_batch_size=1 \
-    --sharding_strategy="hybrid"
+    --sharding_strategy="full" \
```
Comment (sean-smith) [inline 3.test_cases/10.FSDP/1.distributed-training.sbatch:71]: add a comment on why we're making this change

**Review by sean-smith — COMMENTED** [review]
Looks good, let's add a few comments on the changes so we know why we added certain environment variables

**Comment by verdimrc** [comment]
with this PR, llama2-70b can run on two instances (for a total 16x 80GB GPU).

File: 3.test_cases/10.FSDP/model_utils/checkpoint.py [hunk 3]
```diff
@@ -13,34 +13,36 @@
 import torch.distributed as dist
 
 # pylint: disable=import-error,no-name-in-module
-from torch.distributed import checkpoint
+import torch.distributed.checkpoint as dist_cp
+import torch.multiprocessing as mp
```
Comment (pbelevich) [inline 3.test_cases/10.FSDP/model_utils/checkpoint.py:17]:
```suggestion
```
unused?

Comment (johnbensnyder) [reply 3.test_cases/10.FSDP/model_utils/checkpoint.py:17]: Good catch. I was copying the latest PT example, and missed that.

File: 3.test_cases/10.FSDP/1.distributed-training.sbatch (hunk 2 above)
Comment (johnbensnyder) [reply 3.test_cases/10.FSDP/1.distributed-training.sbatch]: This fixes the OOM issues that have been popping up, but sharding across all nodes by default. Added a references in the script to the PT docs.

File: 3.test_cases/10.FSDP/2.distributed-training-mistral.sbatch (hunk 1 above)
Comment (johnbensnyder) [reply 3.test_cases/10.FSDP/2.distributed-training-mistral.sbatch]: Added a link to the Cuda documentation.

**Review by verdimrc — APPROVED** [review]

**Pull request merged**